File: lessons/compiled_lang/java/src/e30.branch.java
import java.io.IOException;

class E30Branch {
    public static void main(String[] args) throws IOException {
        System.out.print(
            "Select an item from the menu.\n\n" +
            "  1 - Coffee\n" +
            "  2 - Espresso\n" +
            "  3 - Latte\n" +
            "  4 - Machiato\n" +
            "  5 - Capucino\n" +
            "  6 - Mocha\n" +
            "  7 - Tea\n\n" +
            "Make your selection: ");

        int selection = System.in.read() - '0';

        if (selection == 1)
            System.out.println("You selected a Coffee");
        else if (selection == 2)
            System.out.println("You selected an Espresso");
        else if (selection == 3)
            System.out.println("You selected a Latte");
        else if (selection == 4)
            System.out.println("You selected a Machiato");
        else if (selection == 5)
            System.out.println("You selected a Capucino");
        else if (selection == 6)
            System.out.println("You selected a Mocha");
        else if (selection == 7)
            System.out.println("You selected a Tea");
        else
            System.out.println("You have not entered a valid selection");
    }
}

File: lessons/compiled_lang/cpp/src/e30.branch.cpp
#include <iostream>

int main() {
    std::cout <<
        "Select an item from the menu.\n\n"
        "  1 - Coffee\n"
        "  2 - Espresso\n"
        "  3 - Latte\n"
        "  4 - Machiato\n"
        "  5 - Capucino\n"
        "  6 - Mocha\n"
        "  7 - Tea\n\n"
        "Make your selection: ";

    char keypress;
    std::cin.get(keypress);
    int selection = keypress - '0';

    if (selection == 1)
        std::cout << "You selected a Coffee" << std::endl;
    else if (selection == 2)
        std::cout << "You selected an Espresso" << std::endl;
    else if (selection == 3)
        std::cout << "You selected a Latte" << std::endl;
    else if (selection == 4)
        std::cout << "You selected a Machiato" << std::endl;
    else if (selection == 5)
        std::cout << "You selected a Capucino" << std::endl;
    else if (selection == 6)
        std::cout << "You selected a Mocha" << std::endl;
    else if (selection == 7)
        std::cout << "You selected a Tea" << std::endl;
    else
        std::cout << "You have not entered a valid selection" << std::endl;

    return 0;
}
